From a0f7205bbe197e61f5cf6b0a9ffdb50ac61dc94b Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 13 Sep 2024 09:32:57 +0300
Subject: [PATCH] [FSW-6580] fs1: set lfs done in recovery entry

With this patch, added support to set lfs done
in recovery entry path. There are controllers
like cxl will check for boot rom done with lfs,
so need to unblock them.

Change-Id: I9ea451edd255d408ae743a9ff4b347fdd46248a0
---

diff --git a/arch/fs1/arc/progress_bits.h b/arch/fs1/arc/progress_bits.h
index 1d58c25..0c0f26d 100644
--- a/arch/fs1/arc/progress_bits.h
+++ b/arch/fs1/arc/progress_bits.h
@@ -152,6 +152,7 @@
 	PROGRESS_BOOT_INFO_LFS_FILE_COPY_PART_FAIL = 135,
 	PROGRESS_BOOT_INFO_LFS_FILE_SET_FAIL = 136,
 	PROGRESS_BOOT_INFO_LFS_FILE_NAME_FAIL = 137,
+	PROGRESS_BOOT_INFO_LFS_FINISHED, /* TODO */
 };
 
 #endif /* _PROGRESS_BITS_H_ */
diff --git a/common/recovery.c b/common/recovery.c
index be9baf8..e548c44 100644
--- a/common/recovery.c
+++ b/common/recovery.c
@@ -58,6 +58,9 @@
 	if (boot_rom_in_multi_die_mode())
 		boot_rom_notify_slave();
 
+	/* TODO: move this flow to master die only */
+	boot_rom_status_update(PROGRESS_BOOT_INFO_LFS_FINISHED);
+
 	if (!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_PRIMARY) &&
 	    !boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_SECONDARY))
 		goto ROM_RECOVERY;
